### DISCUSSION TITLE
如果运营商的IP变了有办法自动更新吗?

### DISCUSSION THREAD

**Discussion by wangx2** [MikeWang000000/Natter#62] [Q&A]
我已经设置了DDNS,域名也可以更新,如果我运行脚本后,除非我自己结束,否则他会一直运行下去

但是运营商的IP是会变化的,每一次变化我都要自己重新运行一下脚本,有没有办法实现自动更新IP以及端口并且推送呢?

**Comment by MikeWang000000** [comment]
不需要重新运行。Natter会自动更新，这是Natter的重要特性之一。
如果遇到IP或端口发生改变，Natter没有及时更新，欢迎汇报Bug。

**Comment by wangx2** [comment]
> 不需要重新运行。Natter会自动更新，这是Natter的重要特性之一。 如果遇到IP或端口发生改变，Natter没有及时更新，欢迎汇报Bug。

好的，我这次看确实更新了。
还有，大佬这个有没有办法实时推送到微信或者TG上？

**Comment by sinaamos** [comment]
> 不需要重新运行。Natter会自动更新，这是Natter的重要特性之一。 如果遇到IP或端口发生改变，Natter没有及时更新，欢迎汇报Bug。

自动端口转发是在hook.sh里设么

**Comment by 1368129224** [comment]
> > 不需要重新运行。Natter会自动更新，这是Natter的重要特性之一。 如果遇到IP或端口发生改变，Natter没有及时更新，欢迎汇报Bug。
> 
> 好的，我这次看确实更新了。 还有，大佬这个有没有办法实时推送到微信或者TG上？

@sinaamos @wangx2 
推送及其他操作可以修改`hook.sh`做到

(毛遂自荐一下，贴了三种模板供参考
https://github.com/1368129224/Natter-User-Guide#%E9%AB%98%E7%BA%A7%E7%94%A8%E6%B3%95
